desktop/desktop_panels.xml

```diff
@@ -464,25 +379,15 @@
 							TableManager.processTableRoll("", "Weather Temperature");
 							TableManager.processTableRoll("", "Weather Precipitation");
 						end
+
 						CalendarManager.adjustMinutes(nMinute);
 						CalendarManager.adjustHours(nHour);
 						CalendarManager.adjustDays(nDay);
 						CalendarManager.adjustMonths(nMonth);
 						CalendarManager.adjustYears(nYear);
 						CalendarManager.outputDate();
 						CalendarManager.outputTime();
-						onUpdate();
-					end
-					local bTimeStep = false;
-					function onUpdate()
-						bTimeStep = true;
-						
-						local nDateinMinutes = TimeManager.getCurrentDateinMinutes();
-						DB.setValue("calendar.dateinminutes", "number", nDateinMinutes); DB.setValue("calendar.dateinminutesstring", "string", tostring(nDateinMinutes))
-
-						local nCurrentRound = DB.getValue("combattracker.round", 0);
-						nCurrentRound = nCurrentRound % 10
-						DB.setValue("combattracker.round", 'number', nCurrentRound);
+						TimeManager.onUpdateAddControl();
```

Comment:
Moved as much of the logic I could to script files.  Easier to maintain and check for errors there.